scripts/gen_gdt.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2017 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import sys
+import struct
+import os
+from elftools.elf.elffile import ELFFile
+from elftools.elf.sections import SymbolTableSection
+
+gdt_pd_fmt = "<HIH"
+
+FLAGS_GRAN = 1 << 7 # page granularity
+ACCESS_EX = 1 << 3  # executable
+ACCESS_DC = 1 << 2  # direction/conforming
+ACCESS_RW = 1 << 1  # read or write permission
+
+# 6 byte pseudo descriptor, but we're going to actually use this as the
+# zero descriptor and return 8 bytes
+def create_gdt_pseudo_desc(addr, size):
+    # ...and take back one byte for the Intel god whose Ark this is...
+    size = size - 1
+    return struct.pack(gdt_pd_fmt, size, addr, 0)
+
+
+# Limit argument always in bytes
+def chop_base_limit(base, limit):
+    base_lo = base & 0xFFFF
+    base_mid = (base >> 16) & 0xFF
+    base_hi = (base >> 24) & 0xFF
+
+    limit_lo = limit & 0xFFFF
+    limit_hi = (limit >> 16) & 0xF
+
+    return (base_lo, base_mid, base_hi, limit_lo, limit_hi)
+
+
+gdt_ent_fmt = "<HHBBBB"
```

Comment:
Style: seems like putting this here is actually less readable than writing the literal into the call to struct.pack() below.  Sure, it's a string literal, but semantically this is "code", not a constant you want to separate for maintainability.